src/katgpucbf/fgpu/engine.py

```diff
@@ -387,6 +387,20 @@ def pols(self) -> int:  # noqa: D401
         """Number of polarisations."""
         return self.spectra.shape[3]
 
+    @property
+    def all_present(self) -> bool:  # noqa: D410
+        """All batches for this chunk are complete.
+
+        :attr:`OutQueueItem.present` does not get "reset" in-between chunks. If
+        one chunk is complete and it is filled with the value `True`, and the
+        last few heaps / batches of the next chunk processed by this Item go
+        missing, you won't be able to tell directly from _out_item.present.
+
+        The canonical way to check whether this item's data is complete is
+        therefore to use this property.
+        """
+        return self.n_spectra == self.capacity and np.all(self.present)  # type: ignore
```

Comment:
Without the `ignore` I got the following:

```
src/katgpucbf/fgpu/engine.py:402: error: Incompatible return value type (got "Literal[False] | bool_", expected "bool")  [return-value]
```

I was too tired to fight with it in this case.

Reply:
np.all returns a numpy bool, but the signature for this function says it returns a Python bool. So try:
```python
        return self.n_spectra == self.capacity and bool(np.all(self.present))
```